numpy/f2py/crackfortran.py

```diff
@@ -2704,18 +2733,15 @@ def analyzevars(block):
             if dim and 'dimension' not in vars[n]:
                 vars[n]['dimension'] = []
                 for d in rmbadname([x.strip() for x in markoutercomma(dim).split('@,@')]):
+                    # d is the expression inside the dimension declaration
                     star = '*'
                     if d == ':':
                         star = ':'
-                    if d in params:
-                        d = str(params[d])
-                    for p in list(params.keys()):
-                        re_1 = re.compile(r'(?P<before>.*?)\b' + p + r'\b(?P<after>.*)', re.I)
-                        m = re_1.match(d)
-                        while m:
-                            d = m.group('before') + \
-                                str(params[p]) + m.group('after')
-                            m = re_1.match(d)
+                    if ((d in params) or
+                            (d.find("(") != -1 and d[:d.find("(")] in params)):
+                        # the dimension for this variable depends on a
+                        # previously defined parameter
+                        d = param_parse(d, params)
```

Comment:
Rather than repeating the logic of `param_parse`, I wonder if this should be
```suggestion
                    try:
                        d = param_parse(d, params)
                    except (ValueError, IndexError, KeyError):
                        pass
```

Reply:
Looks better, thanks!